config_sleep_schedule.go

```diff
@@ -0,0 +1,187 @@
+			"host_allocator_feedback_rule":      c.HostAllocatorFeedbackRule,
+			"hosts_overallocated_rule":          c.HostsOverallocatedRule,
+			"free_host_fraction":                c.FutureHostFraction,
+			"cache_duration_seconds":            c.CacheDurationSeconds,
+			"planner":                           c.Planner,
+			"target_time_seconds":               c.TargetTimeSeconds,
+			"acceptable_host_idle_time_seconds": c.AcceptableHostIdleTimeSeconds,
+			"group_versions":                    c.GroupVersions,
+			"patch_zipper_factor":               c.PatchFactor,
+			"patch_time_in_queue_factor":        c.PatchTimeInQueueFactor,
+			"commit_queue_factor":               c.CommitQueueFactor,
+			"mainline_time_in_queue_factor":     c.MainlineTimeInQueueFactor,
+			"expected_runtime_factor":           c.ExpectedRuntimeFactor,
+			"generate_task_factor":              c.GenerateTaskFactor,
+			"stepback_task_factor":              c.StepbackTaskFactor,
+		},
+	}, options.Update().SetUpsert(true))
+
+	return errors.Wrapf(err, "updating config section '%s'", c.SectionId())
+}
+
+func (c *SchedulerConfig) ValidateAndDefault() error {
+	if c.TaskFinder == "" {
+		// default to legacy
+		c.TaskFinder = FinderVersionLegacy
+	}
+
+	if !utility.StringSliceContains(ValidTaskFinderVersions, c.TaskFinder) {
+		return errors.Errorf("supported task finders are %s; %s is not supported",
+			ValidTaskFinderVersions, c.TaskFinder)
+	}
+
+	if c.HostAllocator == "" {
+		// default to "utilization"
+		c.HostAllocator = HostAllocatorUtilization
+	}
+
+	if !utility.StringSliceContains(ValidHostAllocators, c.HostAllocator) {
+		return errors.Errorf("supported host allocators are %s; %s is not supported",
+			ValidHostAllocators, c.HostAllocator)
+	}
+
+	if c.HostAllocatorRoundingRule == "" {
+		c.HostAllocatorRoundingRule = HostAllocatorRoundDown
+	}
+
+	if !utility.StringSliceContains(ValidDefaultHostAllocatorRoundingRules, c.HostAllocatorRoundingRule) {
+		return errors.Errorf("supported host allocator rounding rules are %s; %s is not supported",
+			ValidDefaultHostAllocatorRoundingRules, c.HostAllocatorRoundingRule)
+	}
+
+	if c.HostAllocatorFeedbackRule == "" {
+		c.HostAllocatorFeedbackRule = HostAllocatorNoFeedback
+	}
+
+	if !utility.StringSliceContains(ValidDefaultHostAllocatorFeedbackRules, c.HostAllocatorFeedbackRule) {
+		return errors.Errorf("supported host allocator feedback rules are %s; %s is not supported",
+			ValidDefaultHostAllocatorFeedbackRules, c.HostAllocatorFeedbackRule)
+	}
+	if c.HostsOverallocatedRule == "" {
+		c.HostsOverallocatedRule = HostsOverallocatedIgnore
+	}
+
+	if !utility.StringSliceContains(ValidDefaultHostsOverallocatedRules, c.HostsOverallocatedRule) {
+		return errors.Errorf("supported hosts overallocation handling rules are %s; %s is not supported",
+			ValidDefaultHostsOverallocatedRules, c.HostsOverallocatedRule)
+	}
+
+	if c.FutureHostFraction < 0 || c.FutureHostFraction > 1 {
+		// traditional default
+		c.FutureHostFraction = .4
```

Comment:
I know this was only moved, but I think it would be nice to have a comment somewhere explaining what this is and why this is the default.

Reply:
Hmm I didn't actually mean to rename the scheduler config file. I swapped the file names so they're correct, so this diff should be gone.